File: ugc/lib/ugc/application.rb

# Licensed to the Apache Software Foundation (ASF) under one or more contributor
# license agreements.  See the NOTICE.txt file distributed with this work for
# additional information regarding copyright ownership.  The ASF licenses this
# file to you under the Apache License, Version 2.0 (the "License"); you may not
# use this file except in compliance with the License.  You may obtain a copy of
# the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
# License for the specific language governing permissions and limitations under
# the License.

module Ugc
  class Application < Usergrid::Application

    def initialize
      super application_url
      self.auth_token = $settings.access_token
    end

    def application_url
      url = $settings.base_url
      org = $settings.organization
      app = $settings.application

      raise "not configured" unless url && org && app

      concat_urls url, "#{org}/#{app}"
    end

    def login(username, password)
      super username, password
      $settings.access_token = auth_token
    end

    def [](uri)
      uri = perform_substitutions uri
      if (URI.parse(uri).host rescue nil)
        Usergrid::Resource.new(uri, nil, $application.options) # absolute
      else
        super # relative
      end
    end

    def list_collections
      app = $application.entity
      collections = app['metadata']['collections']
      table border: $settings.table_border? do
        row header: true do
          collections.first[1].each_key do |k|
            column k
          end
        end
        collections.each_value do |coll|
          row do
            coll.each_value do |v|
              column v
            end
          end
        end
      end
    end
  end
end

File: stack/rest/src/test/java/org/apache/usergrid/rest/test/resource/app/Application.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.usergrid.rest.test.resource.app;


import javax.ws.rs.core.MediaType;

import com.fasterxml.jackson.databind.JsonNode;
import java.io.IOException;
import org.apache.usergrid.rest.test.resource.CustomCollection;
import org.apache.usergrid.rest.test.resource.RootResource;
import org.apache.usergrid.rest.test.resource.ValueResource;
import org.apache.usergrid.rest.test.resource.app.queue.DevicesCollection;
import org.apache.usergrid.rest.test.resource.app.queue.QueuesCollection;


/** @author tnine */
public class Application extends ValueResource {


    /**
     * @param parent
     */
    public Application( String orgName, String appName, RootResource root ) {
        super( orgName + SLASH + appName, root );
    }


    /** Get the token from management for this username and password */
    public String token( String username, String password ) throws IOException {

        String url = String.format( "%s/token", url() );

        JsonNode node = mapper.readTree( resource().path( url ).queryParam( "grant_type", "password" ).queryParam( "username", username )
                .queryParam( "password", password ).accept( MediaType.APPLICATION_JSON )
                .type( MediaType.APPLICATION_JSON_TYPE ).get( String.class ));

        return node.get( "access_token" ).asText();
    }


    public UsersCollection users() {
        return new UsersCollection( this );
    }

    public GroupsCollection groups() {
        return new GroupsCollection( this );
    }

    public RolesCollection roles() {
        return new RolesCollection( this );
    }


    public QueuesCollection queues() {
        return new QueuesCollection( this );
    }


    public DevicesCollection devices() {
        return new DevicesCollection( this );
    }

//TODO: work out differences between CustomCollections and replace tests with a general collection method.
    public CustomCollection customCollection( String name ) {
        return new CustomCollection( name, this );
    }

    public Collection collection( String name ) {
        return new Collection( name,this );
    }
}
